dotnetcore2.0/run/MockBootstraps/Program.cs

```diff
@@ -9,50 +10,79 @@ namespace MockLambdaRuntime
+                var shouldWaitForDebugger = GetShouldWaitForDebuggerFlag(args, out var positionalArgs);
+
+                var handler = GetFunctionHandler(positionalArgs);
+                var body = GetEventBody(positionalArgs);
+
+                if (shouldWaitForDebugger)
+                {
+                    Console.Error.WriteLine("Waiting for the debugger to attach...");
+
+                    if (!DebuggerExtensions.TryWaitForAttaching(
+                        _debuggerStatusQueryInterval,
+                        _debuggerStatusQueryTimeout))
+                    {
+                        Console.Error.WriteLine("Timeout. Proceeding without debugger.");
+                    }
+                }
+
+                var lambdaContext = new MockLambdaContext(handler, body);
+
+                var userCodeLoader = new UserCodeLoader(handler, InternalLogger.NO_OP_LOGGER);
+                userCodeLoader.Init(Console.Error.WriteLine);
+
+                var lambdaContextInternal = new LambdaContextInternal(lambdaContext.RemainingTime,
+                                                                      LogAction, new Lazy<CognitoClientContextInternal>(),
+                                                                      lambdaContext.RequestId,
+                                                                      new Lazy<string>(lambdaContext.Arn),
+                                                                      new Lazy<string>(string.Empty),
+                                                                      new Lazy<string>(string.Empty),
+                                                                      Environment.GetEnvironmentVariables());
+
+                Exception lambdaException = null;
+
+                LogRequestStart(lambdaContext);
+                try
+                {
+                    userCodeLoader.Invoke(lambdaContext.InputStream, lambdaContext.OutputStream, lambdaContextInternal);
+                }
+                catch (Exception ex)
+                {
+                    lambdaException = ex;
+                }
+                LogRequestEnd(lambdaContext);
+
+                if (lambdaException == null)
+                {
+                    Console.WriteLine(lambdaContext.OutputText);
+                }
+                else
+                {
+                    Console.Error.WriteLine(lambdaException);
+                }
             }
-            catch (Exception ex)
-            {
-                lambdaException = ex;
-            }
-            LogRequestEnd(lambdaContext);
 
-            if (lambdaException == null)
-            {
-                Console.WriteLine(lambdaContext.OutputText);
-            }
-            else
+            // Catch all unhandled exceptions from runtime, to prevent user from hanging on them while debugging
+            catch (Exception ex)
             {
-                Console.Error.WriteLine(lambdaException);
```

Comment:
I don't understand this change

Reply:
I've added `"\nUnhandled exception occured in runner:\n"` to better constraint the exception origin for the user, while he is debugging. And outer `try/catch` block is used to prevent user from hanging on unhandled exception from the runnner program.

@mhart does it make sense?

Reply:
I don't think so – I think it'd be better to match the behaviour of the rest of the runtimes w/ regards to error output. So errors should look the same as they do on Lambda itself.

If it doesn't currently do that (it might not by the looks of it) – then we should change it so that it does 👍

Reply:
![issue](https://user-images.githubusercontent.com/34073648/49391655-f2e11f00-f734-11e8-9a41-480584acee7b.gif)

On this sample I've misused Lambda runner and got stuck on unhandled exception (default dotnet debugger behavior), that is why I have `try/catch` block around all the invocation.

@mhart Did I made my point clear?

Reply:
To further expand on the topic, this `try/catch` only handles errors in runner and not the Lambda itself. So behavior regarding Lambda error output will not change in any way. This will only prevent unexpected user hanging while debugging

Reply:
Gotcha, I see what you mean – will have a play with this and see if the extra explanation string is useful.

Reply:
Okay, let me know, what you think, @mhart. 
And in general you are okay with the flag, so I can start working on SAM CLI PR in the meantime which relies on this implementation?

Reply:
That's a good point – maybe we should make it something quite specific so that it doesn't clash with any potential future flags that AWS might add to Lambda. Something like `--wait-for-debugger` or `--debug-wait`

Reply:
How about `--debugger-spin-wait` ?

Reply:
Yup, happy with that! 👍

Reply:
Updated according to comments

Reply:
Great, I'm happy with that, feel free to update your SAM CLI PR 👍 

I'll explore this PR more in the next couple of days, but I think the rest of it looks pretty good

Reply:
Great, take your time @mhart